src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php

```diff
@@ -672,11 +672,11 @@ public function createOrFirst(array $attributes = [], array $values = [], array
      */
     public function updateOrCreate(array $attributes, array $values = [], array $joining = [], $touch = true)
     {
-        return tap($this->firstOrCreate($attributes, $values), function ($instance) use ($values) {
+        return tap($this->firstOrCreate($attributes, $values, $joining, $touch), function ($instance) use ($values, $touch) {
             if (! $instance->wasRecentlyCreated) {
                 $instance->fill($values);
 
-                $instance->save(['touch' => false]);
+                $instance->save(compact('touch'));
```

Comment:
Is it safe to pass the `$touch` argument to both `firstOrCreate` and `save`? If the timestamp update process runs twice, how should we handle it...

Reply:
That was a mistake, okay